extensions/cl_khr_unified_svm.asciidoc

```diff
@@ -0,0 +1,1082 @@
+|====
+| Device Info | Return Type | Description
+| `CL_DEVICE_SVM_TYPE_CAPABILITIES_KHR`
+  | `cl_svm_capabilities_khr[]`
+      | Queries the combinations of SVM capabilities describing the SVM types supported by an OpenCL device.
+        Returns an array of bitfields, where each bitfield in the array describes the SVM capabilities for one SVM type.
+        The size of the returned array must match the size of the array returned by the platform query `CL_PLATFORM_SVM_TYPE_CAPABILITIES_KHR`.
+
+        Each entry in the returned array must be either a super-set of the entry in the array returned by the platform query `CL_PLATFORM_SVM_TYPE_CAPABILITIES_KHR`, indicating that the SVM type is supported by the device, or zero, indicating that the SVM type is not supported by this device.
+
+        Please refer to the <<svm-capabilities-table,SVM capabilities>> table for valid capability values and their description.
+|====
+
+[[svm-capabilities-table]]
+[caption="Table X. "]
+.List of SVM capabilities
+[width="100%",cols="2,3",options="header"]
+|====
+| Capability | Description
+| `CL_SVM_CAPABILITY_SINGLE_ADDRESS_SPACE_KHR`
+  | There is a single address space for this type of SVM.
+    The same pointer may be used on the host and the device; the pointer has _address equivalence_.
+| `CL_SVM_CAPABILITY_SYSTEM_ALLOCATED_KHR`
+  | This type of SVM provides access to the entire host virtual memory, including memory allocated by a system allocator such as `malloc` or `new` or objects allocated on the stack, and does not require calling *clSVMAllocWithPropertiesKHR* or *clSVMAlloc*.
+| `CL_SVM_CAPABILITY_DEVICE_OWNED_KHR`
+  | This type of SVM is owned by an associated device handle and is not intended to migrate to another device or the host.
+    Allocations that are owned by a device generally trade off access limitations for higher performance.
+| `CL_SVM_CAPABILITY_DEVICE_UNASSOCIATED_KHR`
+  | This type of SVM does not need to be associated with a device handle.
+| `CL_SVM_CAPABILITY_CONTEXT_ACCESS_KHR`
+  | This type of SVM is accessible to other devices in the context that support the SVM type.
+| `CL_SVM_CAPABILITY_HOST_OWNED_KHR`
+  | This type of SVM is owned by the host and is not intended to migrate to a device.
+    Allocations that are owned by the host generally trade off wide accessibility for potentially higher per-access costs.
+| `CL_SVM_CAPABILITY_HOST_READ_KHR`
+  | This type of SVM is readable on the host without needing to map or unmap the allocation.
+| `CL_SVM_CAPABILITY_HOST_WRITE_KHR`
+  | This type of SVM is writeable on the host without needing to map or unmap the allocation.
+| `CL_SVM_CAPABILITY_HOST_MAP_KHR`
+  | This type of SVM is accessible on the host but requires mapping and unmapping the allocation.
+| `CL_SVM_CAPABILITY_DEVICE_READ_KHR`
+  | This type of SVM is accessible on the device for reading.
+| `CL_SVM_CAPABILITY_DEVICE_WRITE_KHR`
+  | This type of SVM is writeable on the device for writing.
```

Comment:
We should clarify exactly what is meant by "accessible on the device for reading" and "accessible on the device for writing", given issues like #770.

* Do these capabilities only apply to kernels, or do they apply to other APIs as well?

Specifically:
* What capabilities are needed for clEnqueueSVMMemcpy?  Does it matter if the SVM allocation is the memcpy source or destination?
* What capabilities are needed for clEnqueueSVMMemFill?
* What capabilities are needed for clEnqueueSVMMigrateMem?

Minor, there is a small typo here too: "writeable on the device for writing" should be "accessible on the device for writing".